### PR TITLE
Add 3.1 and 3.2 to CI and supported versions

### PR DESCRIPTION
**Pull request by nirebu**
What the title says, if it's desirable.

### CODE REVIEW

**Comment by nirebu** [comment]
@michaelherold I can help you with that. I think I'll try on a different repo where I can trigger actions without waiting for your input, then transfer the results here.

**Comment by nirebu** [comment]
@michaelherold I've updated the CI file to use the non-deprecated action.

I've also added [7a36b95](https://github.com/michaelherold/benchmark-memory/pull/25/commits/7a36b959f29537564a6d29838079d6454e367a62) because specs were breaking on 3.0 and 3.2 because of the old `pry` version. Not sure why they were not failing on 3.1 though.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -48,11 +48,13 @@ jobs:
           - 2.6
           - 2.7
           - 3.0
```
Comment (AlexWayfer) [inline .github/workflows/ci.yaml:50]: You have to take versions in quoutes: https://github.com/ruby/setup-ruby#matrix-of-ruby-versions

> Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'

Comment (nirebu) [reply .github/workflows/ci.yaml]: @AlexWayfer done! Didn't notice it. Thanks for pointing this out.

**Review by AlexWayfer — APPROVED** [review]
Thanks.

**Comment by michaelherold** [comment]
Thanks!

**Pull request merged**